xbmc/input/GamepadTranslator.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ *      Copyright (C) 2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "GamepadTranslator.h"
+#include "Key.h"
+#include "utils/log.h"
+#include "utils/StringUtils.h"
+
+#include <string>
+
+uint32_t CGamepadTranslator::TranslateGamepadString(const char *szButton)
```

Comment:
since such a generic name can cause issues remember the namespace NAMESPACE_WHERE_WE_PUT_INPUT_FROM_GAMEPAD_AND_OTHER_STUFFS. and name the method DoThatThingWhereAStringIsMadeDifferent1() for clarity.